feat: add mongodb atlas docs for metrics and logs (#1717)
SigNoz/signoz.io: data/docs/integrations/mongodb-atlas.mdx

---
date: 2024-08-03
id: mongodb-atlas
title: MongoDB Atlas Metrics and Logs
---

## Overview

> This integration helps you monitor key MongoDB Atlas metrics and logs, view them with an out-of-the-box dashboard, and parse MongoDB Atlas logs for better querying and aggregation.

Reviewer: ```suggestion
MongoDB Atlas is a fully managed cloud database service for modern applications. This documentation guides you through setting up monitoring for MongoDB Atlas—tracking key metrics, collecting logs, and visualising them through an out-of-the-box dashboard.
```

> The MongoDB Atlas receiver collects OpenTelemetry data from MongoDB Atlas, including database performance metrics, operational data, audit logs, process metrics and system events.

Reviewer: ```suggestion
```

## Prerequisites

Before you begin, ensure you have:

1. A MongoDB Atlas account with appropriate access
2. MongoDB Atlas API keys with required permissions:
   - **Public Key** and **Private Key** with access for the corresponding MongoDB Atlas Project
   - **Project Data Access Read Only** role (minimum) for accessing project events and logs

> 3. For audit logs: Audit logging must be enabled in your MongoDB Atlas project

Reviewer: Hyeprlink
```suggestion
3. For audit logs: Audit logging [must be enabled](https://www.mongodb.com/docs/atlas/database-auditing/) in your MongoDB Atlas project
```

> ## Collecting MongoDB Atlas Logs and Metrics

Reviewer: ```suggestion
## Setup
```

<Tabs>
<TabItem value="signoz-cloud" label="SigNoz Cloud" default>
## Collecting MongoDB Atlas Logs

### Setup SigNoz OTel Collector

### Step 1: Install OpenTelemetry Collector

> Follow the [OpenTelemetry SigNoz document](https://signoz.io/docs/tutorial/opentelemetry-binary-usage-in-virtual-machine/) to install the OpenTelemetry Collector.

Reviewer: ```suggestion
Follow the [this document](https://signoz.io/docs/tutorial/opentelemetry-binary-usage-in-virtual-machine/) to install the OpenTelemetry Collector.
```

Make this change everywhere we say `OpenTelemetry SigNoz document`

<figure data-zoomable align='center'>
    <img src="/img/docs/integrations/mongodb-atlas/logs.webp" alt="MongoDB Atlas Logs"/>
    <figcaption><i>MongoDB Atlas Logs in SigNoz</i></figcaption>
</figure>

<figure data-zoomable align='center'>
    <img src="/img/docs/integrations/mongodb-atlas/detailed-log.webp" alt="MongoDB Atlas Detailed Log"/>
    <figcaption><i>MongoDB Atlas Detailed Log in SigNoz</i></figcaption>
</figure>

### MongoDB Atlas Dashboard

You can import the MongoDB Atlas dashboard for monitoring your MongoDB Atlas metrics.

Go to Dashboards => New Dashboard => Import JSON

Download JSON from [link](https://github.com/SigNoz/dashboards/blob/main/mongodb-atlas/mongodb-atlas-otlp-v1.json).

Here are sample screenshots from the dashboard:

<figure data-zoomable align='center'>
    <img src="/img/docs/integrations/mongodb-atlas/overview.webp" alt="MongoDB Atlas Overview"/>
    <figcaption><i>MongoDB Atlas Overview</i></figcaption>
</figure>

<figure data-zoomable align='center'>
    <img src="/img/docs/integrations/mongodb-atlas/project-stats.webp" alt="MongoDB Atlas Project Stats"/>
    <figcaption><i>MongoDB Atlas Project Stats</i></figcaption>
</figure>

> <details> 

Reviewer: This drop-down should be part of the metrics tab for both self-host and cloud

| mongodbatlas.system.paging.usage.max                                      | Gauge | KiloBytes            | Swap usage                                                                                              |
| mongodbatlas.disk.partition.queue.depth                                   | Gauge | number               | Disk queue depth                                                                                        |
| mongodbatlas.disk.partition.throughput                                    | Gauge | Bytes/s              | Disk throughput                                                                                         |
| mongodbatlas.process.cache.ratio                                          | Gauge | number               | Cache ratios represented as (%)                                                                         |

You can refer [this](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/mongodbatlasreceiver/documentation.md) page for more details about the MongoDB Atlas metrics.

</details>

## Troubleshooting

> ### Logs Collection Issues

Reviewer: Add some troubleshooting for metrics collection as well.
